core/src/main/java/com/datastrato/gravitino/storage/relational/converters/SQLExceptionConverterFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.storage.relational.converters;
+
+import com.datastrato.gravitino.Config;
+import com.datastrato.gravitino.Configs;
+import com.google.common.base.Preconditions;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SQLExceptionConverterFactory {
+  private static final Pattern TYPE_PATTERN = Pattern.compile("jdbc:(\\w+):");
+  private static SQLExceptionConverter converter;
+
+  private SQLExceptionConverterFactory() {}
+
+  public static synchronized void initConverter(Config config) {
```

Comment:
When will this method be accessed concurrently?

Reply:
There is currently no concurrent access, just ensure the singleton of the `converter` in the factory class.

Reply:
OK